### PR TITLE
Fix #3077: Do not error on scorer generation when no error was generated

### LINKED ISSUE #3077: generate_lm.py errors on success
reported on discourse: https://discourse.mozilla.org/t/error-when-creating-my-own-scorer-file/62139

likely because of the error checking with an older version (stable, 0.7.3) of the ctc decoder package since we have not pushed a new release

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

File: data/lm/generate_package.py [hunk 1]
```diff
@@ -62,7 +62,8 @@ def create_bundle(
     scorer.fill_dictionary(list(words))
     shutil.copy(lm_path, package_path)
     # append, not overwrite
-    if scorer.save_dictionary(package_path, True):
+    save_dic = scorer.save_dictionary(package_path, True)
+    if save_dic is None or save_dic:
```
Comment (reuben) [inline data/lm/generate_package.py:66]: Why should we support multiple versions of the Scorer code here? We've never done this, instead instructing users to install matching versions of packages/code/models.

Comment (reuben) [reply data/lm/generate_package.py:66]: Maybe we should make generate_package.py run some version checks like we did in train.py.

Comment (lissyx) [reply data/lm/generate_package.py:66]: > Why should we support multiple versions of the Scorer code here? We've never done this, instead instructing users to install matching versions of packages/code/models.

We have a misleading error message here. Do you feel confident releasing a 0.7.4 with all the changes that happened? That's the only correct solution, I agree.

Comment (reuben) [reply data/lm/generate_package.py:66]: We only have a misleading error message if the user is using mixed versions of generate_package.py and ds-ctcdecode package, no?

Comment (lissyx) [reply data/lm/generate_package.py:66]: > We only have a misleading error message if the user is using mixed versions of generate_package.py and ds-ctcdecode package, no?

Yes, which is the case if people do follow docs on current master, so we'll get people complaining and adding noise here :/

**Pull request closed without merging**